rust/telemetry-sink/src/http_event_sink.rs

```diff
@@ -91,26 +92,32 @@ impl HttpEventSink {
         client: &mut reqwest::Client,
         root_path: &str,
         process_info: Arc<ProcessInfo>,
-        retry_strategy: core::iter::Take<tokio_retry::strategy::ExponentialBackoff>,
+        retry_strategy: core::iter::Take<tokio_retry2::strategy::ExponentialBackoff>,
         decorator: &dyn RequestDecorator,
     ) -> Result<()> {
         debug!("sending process {process_info:?}");
         let url = format!("{root_path}/ingestion/insert_process");
-        tokio_retry::Retry::spawn(retry_strategy, || async {
+        tokio_retry2::Retry::spawn(retry_strategy, || async {
             let body = encode_cbor(&*process_info)?;
-            let mut request = client.post(&url).body(body).build()?;
+            let mut request = client
+                .post(&url)
+                .body(body)
+                .build()
+                .with_context(|| "building request")
+                .map_transient()?;
             decorator
                 .decorate(&mut request)
                 .await
-                .with_context(|| "decorating request")?;
+                .with_context(|| "decorating request")
+                .map_transient()?;
```

Comment:
what is map_transient?

Reply:
It is a wrapper for map_err(RetryError::transient). If you use RetryError::permanent it breaks the retry cycle. Inspired by back off crate

Reply:
so it looks like transient errors can be retried. If it's not explicitly marked as transient, what happens? Does it not compile or is it implicitly permanent?

Reply:
Exactly, for most errors it was mapping correctly, but not here. Transient is the default behaviour

Reply:
I have created a new branch that tests implicit conversions to avoid this. Will test it tomorrow